bigframes/core/indexers.py

```diff
@@ -391,6 +394,23 @@ def _perform_loc_list_join(
     return result
 
 
+def _struct_accessor_check_and_warn(
+    series: bigframes.series.Series, index: indexes.Index
+):
+    if not bigframes.dtypes.is_struct_like(series.dtype):
+        # No need to check series that do not have struct values
+        return
+
+    if not bigframes.dtypes.is_string_like(index.dtype):
+        # No need to check indexing with non-string values.
+        return
+
+    if not bigframes.dtypes.is_string_like(series.index.dtype):
+        warnings.warn(
+            "Are you trying to access struct fields? If so, please use Series.struct.field(...) method instead."
```

Comment:
Could we do a custom category (defined in bigframes.exceptions) here to make it a bit easier to filter out if needed for some reason?

Also, I recommend setting stacklevel so that the error shows up in user code, not as appearing deep in BigFrames.

Reply:
I have provided a stack level such that the warning call site is now at `__getitem__`